src/test/mocks/vaults.ts

```diff
@@ -0,0 +1,42 @@
+import { bnOrZero } from '@shapeshiftoss/chain-adapters'
+import { ChainTypes } from '@shapeshiftoss/types'
+import { MergedEarnVault } from 'pages/Defi/hooks/useVaultBalances'
+
+import { YearnVaultWithApyAndTvl } from '../../hooks/useVaultWithoutBalance/useVaultWithoutBalance'
+
+export const mockVault = (obj?: {
+  vaultAddress?: string
+  underlyingTokenBalanceUsdc?: string
+  apy?: number
+}) =>
+  ({
+    vaultAddress: '',
+    underlyingTokenBalanceUsdc: '',
+    apy: 0,
+    chain: ChainTypes.Ethereum,
+    name: '',
+    symbol: '',
+    tokenAddress: '',
+    provider: '',
+    type: '',
+    ...obj
+  } as YearnVaultWithApyAndTvl)
```

Comment:
Can you change the return type of the function here to avoid casting? Casting is always a code smell.

Reply:
Done ! thank you :)